libr/bin/format/elf/elf.c

```diff
@@ -549,157 +530,224 @@ static void set_default_value_dynamic_info(ELFOBJ *bin) {
+			bin->dyn_info.dt_pltrelsz = d.d_un.d_val;
 			break;
 		case DT_PLTGOT:
-			bin->dyn_info.dt_pltgot = dyn[i].d_un.d_ptr;
+			bin->dyn_info.dt_pltgot = d.d_un.d_ptr;
 			break;
 		case DT_HASH:
-			bin->dyn_info.dt_hash = dyn[i].d_un.d_ptr;
+			bin->dyn_info.dt_hash = d.d_un.d_ptr;
 			break;
 		case DT_STRTAB:
-			strtabaddr = Elf_(r_bin_elf_v2p_new) (bin, dyn[i].d_un.d_ptr);
-			bin->dyn_info.dt_strtab = dyn[i].d_un.d_ptr;
+			bin->dyn_info.dt_strtab = d.d_un.d_ptr;
 			break;
 		case DT_SYMTAB:
-			bin->dyn_info.dt_symtab = dyn[i].d_un.d_ptr;
+			bin->dyn_info.dt_symtab = d.d_un.d_ptr;
 			break;
 		case DT_RELA:
-			bin->dyn_info.dt_rela = dyn[i].d_un.d_ptr;
+			bin->dyn_info.dt_rela = d.d_un.d_ptr;
 			break;
 		case DT_RELASZ:
-			bin->dyn_info.dt_relasz = dyn[i].d_un.d_val;
+			bin->dyn_info.dt_relasz = d.d_un.d_val;
 			break;
 		case DT_RELAENT:
-			bin->dyn_info.dt_relaent = dyn[i].d_un.d_val;
+			bin->dyn_info.dt_relaent = d.d_un.d_val;
 			break;
 		case DT_STRSZ:
-			strsize = dyn[i].d_un.d_val;
-			bin->dyn_info.dt_strsz = dyn[i].d_un.d_val;
+			bin->dyn_info.dt_strsz = d.d_un.d_val;
 			break;
 		case DT_SYMENT:
-			bin->dyn_info.dt_syment = dyn[i].d_un.d_val;
+			bin->dyn_info.dt_syment = d.d_un.d_val;
 			break;
 		case DT_REL:
-			bin->dyn_info.dt_rel = dyn[i].d_un.d_ptr;
+			bin->dyn_info.dt_rel = d.d_un.d_ptr;
 			break;
 		case DT_RELSZ:
-			bin->dyn_info.dt_relsz = dyn[i].d_un.d_val;
+			bin->dyn_info.dt_relsz = d.d_un.d_val;
 			break;
 		case DT_RELENT:
-			bin->dyn_info.dt_relent = dyn[i].d_un.d_val;
+			bin->dyn_info.dt_relent = d.d_un.d_val;
 			break;
 		case DT_PLTREL:
-			bin->dyn_info.dt_pltrel = dyn[i].d_un.d_val;
+			bin->dyn_info.dt_pltrel = d.d_un.d_val;
 			break;
 		case DT_JMPREL:
-			bin->dyn_info.dt_jmprel = dyn[i].d_un.d_ptr;
+			bin->dyn_info.dt_jmprel = d.d_un.d_ptr;
 			break;
 		case DT_MIPS_PLTGOT:
-			bin->dyn_info.dt_mips_pltgot = dyn[i].d_un.d_ptr;
+			bin->dyn_info.dt_mips_pltgot = d.d_un.d_ptr;
 			break;
 		case DT_BIND_NOW:
 			bin->dyn_info.dt_bind_now = true;
 			break;
 		case DT_FLAGS:
-			bin->dyn_info.dt_flags = dyn[i].d_un.d_val;
+			bin->dyn_info.dt_flags = d.d_un.d_val;
 			break;
 		case DT_FLAGS_1:
-			bin->dyn_info.dt_flags_1 = dyn[i].d_un.d_val;
+			bin->dyn_info.dt_flags_1 = d.d_un.d_val;
 			break;
 		case DT_RPATH:
-			bin->dyn_info.dt_rpath = dyn[i].d_un.d_val;
+			bin->dyn_info.dt_rpath = d.d_un.d_val;
 			break;
 		case DT_RUNPATH:
-			bin->dyn_info.dt_runpath = dyn[i].d_un.d_val;
+			bin->dyn_info.dt_runpath = d.d_un.d_val;
+			break;
+		case DT_NEEDED:
+			bin->dyn_info.dt_needed[j] = d.d_un.d_val;
+			j++;
 			break;
 		default:
-			if ((dyn[i].d_tag >= DT_VERSYM) && (dyn[i].d_tag <= DT_VERNEEDNUM)) {
-				bin->version_info[DT_VERSIONTAGIDX (dyn[i].d_tag)] = dyn[i].d_un.d_val;
+			if ((d.d_tag >= DT_VERSYM) && (d.d_tag <= DT_VERNEEDNUM)) {
+				bin->version_info[DT_VERSIONTAGIDX (d.d_tag)] = d.d_un.d_val;
 			}
 			break;
```

Comment:
could you put a `eprintf("Dynamic tag %d not handled", dyn[i].d_tag);` or something like that?

Reply:
A lot of test fail because of the warning,
Imho it is an important message.

Reply:
I see... Well, so far I see only 4 tests failing because of this, but I still think it would be good for users (and devs) to know some stuff are not well identified. @kazarmy what do you think?